### PR TITLE
CI cleanups

### PR DESCRIPTION
**Pull request by rathorevaibhav**
## Summary

1. Removed duplicate variables
2. Removed GitHub environment. Directly picking from organization env variables

## Test Plan

- [ ] Test Responsiveness
   - [ ] Laptop (1200px)
   - [ ] Tablet (760px)
   - [ ] Phone (320px)
- [ ] Cross-Browser Testing
   - [ ] Chrome
   - [ ] Firefox
   - [ ] Safari (use BrowserStack)
- [ ] Local Language Support
- [ ] Hygiene and Housekeeping
   - [ ] Self-review
   - [ ] Comments have been added appropriately
   - [ ] Check for bundle size [here](https://bundlephobia.com/) if adding a package
   - [ ] Added relevant details like Labels/Projects/Milestones etc.
- [ ] Testing
   - [ ] Wrote tests
   - [ ] Tested locally
   - [ ] Tested on staging
   - [ ] Tested on an actual physical phone
   - [ ] Tested on production
- [ ] Lighthouse Checks
   - [ ] Images have `alt` attributes
   - [ ] Any `` tags have `width` and `height` specified
   - [ ] Any `target="_blank"` links have `rel="noopener"`
   - [ ] Only SVGs are used as images. If PNGs are used, their size has been optimised.
   - [ ] Any SVGs without text have their `aria-label` attributes set

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -90,6 +106,16 @@ jobs:
...
+          # check branch and switch to branch if exists
+          if [ `git branch --list --remote origin/${{ github.head_ref }}` ]
+          then
+            echo "Switching to branch ${{ github.head_ref }}."
+            git checkout ${{ github.head_ref }}
+            git pull origin ${{ github.head_ref }}
+          else
+            echo "Branch not found. Going with default branch."
+          fi
+
```
Comment (dalmia) [inline .github/workflows/ci.yml:118]: same comment on reusing the output of this step as mentioned in [avantifellows/plio-analytics#26](https://github.com/avantifellows/plio-analytics/pull/26)

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -51,13 +47,23 @@ jobs:
           git clone https://github.com/avantifellows/plio-frontend/
           cd plio-frontend/
 
+          # check branch and switch to branch if exists
+          if [ `git branch --list --remote origin/${{ github.head_ref }}` ]
```
Comment (deepansh96) [inline .github/workflows/ci.yml:51]: Screenshot 2021-12-07 at 12 54 03 PM

Here, the `head_ref` will only be available if this workflow is being run on a pull request. What happens when we merge to prod, will the cypress CI runs won't happen?

It's another story if we don't want them to run when merging branches.

Comment (deepansh96) [reply .github/workflows/ci.yml:51]: I'm guessing during pushing to main/prod, as the `head_ref` will not be available, the code will just pull the default branch, right?

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -51,13 +47,23 @@ jobs:
...
           # create the env file
           cp .env.example .env
 
           # add env secrets
-          echo 'VUE_APP_GOOGLE_CLIENT_ID=${{ secrets.VUE_APP_GOOGLE_CLIENT_ID }}' >> .env
-          echo 'VUE_APP_BACKEND_API_CLIENT_ID=${{ secrets.VUE_APP_BACKEND_API_CLIENT_ID }}' >> .env
-          echo 'VUE_APP_BACKEND_API_CLIENT_SECRET=${{ secrets.VUE_APP_BACKEND_API_CLIENT_SECRET }}' >> .env
+          echo 'VUE_APP_GOOGLE_CLIENT_ID=${{ secrets.GOOGLE_OAUTH2_CLIENT_ID }}' >> .env
+          echo 'VUE_APP_BACKEND_API_CLIENT_ID=${{ secrets.DJANGO_DEFAULT_OAUTH2_CLIENT_ID }}' >> .env
+          echo 'VUE_APP_BACKEND_API_CLIENT_SECRET=${{ secrets.DJANGO_DEFAULT_OAUTH2_CLIENT_SECRET }}' >> .env
```
Comment (deepansh96) [inline .github/workflows/ci.yml:66]: Were these changed in the github secrets as well? And why was this change needed?

**Review by deepansh96 — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 2 above)
Comment (rathorevaibhav) [reply .github/workflows/ci.yml]: That's a good point. I just checked and it seems to be returning empty value in case of direct push to the branch (without a pull request).

Since because of the empty value, the if condition will be false as it won't find a valid remote branch, hence it will go with default, which in our case is master. So I believe you're right that it will still run successfully.

File: .github/workflows/ci.yml (hunk 3 above)
Comment (rathorevaibhav) [reply .github/workflows/ci.yml:66]: @deepansh96 I've updated documentation for Cypress CI. Please have a look.

File: docs/CYPRESS.md [hunk 4]
```diff
@@ -66,27 +66,28 @@ The Cypress Dashboard is a web interface that allows you to run and debug your C
...
-      2. `VUE_APP_BACKEND_API_CLIENT_SECRET`
-      3. `VUE_APP_GOOGLE_CLIENT_ID`
-   5. Plio Backend (Django) - Refer [Plio Backend ENV guide](https://github.com/avantifellows/plio-backend/blob/master/docs/ENV.md)
-      1. `DJANGO_DEFAULT_OAUTH2_CLIENT_ID`
-      2. `DJANGO_DEFAULT_OAUTH2_CLIENT_SECRET`
-      3. `DJANGO_DEFAULT_OAUTH2_CLIENT_SETUP`
-      4. `DJANGO_GOOGLE_OAUTH2_CLIENT_ID`
-      5. `DJANGO_GOOGLE_OAUTH2_CLIENT_SECRET`
-      6. `DJANGO_SECRET_KEY`
+The job `integration-tests` inside the GitHub workflow file [ci.yml](../.github/workflows/ci.yml) is used to run the integration tests in GitHub Actions. Since Cypress runs in multiple Plio repositories, it's optimal to go with organization secrets that can be shared across several repositories. This saves time and effort to create env variables in every repository.
```
Comment (deepansh96) [inline docs/CYPRESS.md:69]:
```suggestion
The job `integration-tests` inside the GitHub workflow file [ci.yml](../.github/workflows/ci.yml) is used to run the integration tests in GitHub Actions. Since Cypress runs require access to multiple Plio repositories, it's optimal to go with organization secrets that can be shared across several repositories. This saves time and effort to create env variables in every repository.
```

**Review by deepansh96 — CHANGES_REQUESTED** [review]

File: docs/CYPRESS.md (hunk 4 above)
Comment (dalmia) [inline docs/CYPRESS.md:69]:
```suggestion
The job `integration-tests` inside the GitHub workflow file [ci.yml](../.github/workflows/ci.yml) is used to run the integration tests in GitHub Actions. Since Cypress runs require setting up multiple Plio repositories, it's optimal to go with organization secrets that can be shared across several repositories. This saves time and effort to create env variables in every repository.
```

**Comment by dalmia** [comment]
Screenshot 2021-12-07 at 3 02 47 PM
